### PR TITLE
New package: mstflint-4.11.0.3

### PR DESCRIPTION
**Pull request by ArmedAviator**
Very useful for Mellanox users.  Successfully used this to flash new firmware on my Mellanox 10G NIC.

### CODE REVIEW

**Comment by newbluemoon** [comment]
This needs #9094 first.

File: srcpkgs/mstflint/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'mstflint'
+pkgname=mstflint
+version=4.11.0.3
+revision=1
+wrksrc="${pkgname}-4.11.0"
+build_style=gnu-configure
+configure_args="--enable-fw-mgr --enable-openssl"
+makedepends="boost-devel libcurl-devel libibmad-devel liblzma-devel libressl-devel libxml2-devel rdma-core-devel zlib-devel"
```
Comment (maxice8) [inline srcpkgs/mstflint/template]: cut at 80 columns

File: srcpkgs/mstflint/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+depends="libboost_regex1.69 libcurl libibmad liblzma libxml2 rdma-core zlib"
```
Comment (maxice8) [inline srcpkgs/mstflint/template]: remove the ones that are automatically detected via shlibs

File: srcpkgs/mstflint/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+short_desc="Mellanox firmware burning and diagnostics tools"
+maintainer="Rich Gannon <rich@richgannon.net>"
+license="GPL-2.0-or-later"
+homepage="https://github.com/Mellanox/mstflint"
+distfiles="${homepage}/releases/download/v4.11.0-3/mstflint-4.11.0-3.tar.gz"
```
Comment (maxice8) [inline srcpkgs/mstflint/template]: use ${version} and write out the homepage varaible

File: srcpkgs/mstflint/template [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+checksum=ede5f897b5a772ef35faa49a94517059510764b3d50df52b267555e17c0c6594
+
+mstflint-devel_package() {
+	depends="mstflint-${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/libmstarchive.a"
```
Comment (maxice8) [inline srcpkgs/mstflint/template]: no need to quote if you are not using wildcard (*)

File: srcpkgs/mstflint/template (hunk 3 above)
Comment (maxice8) [reply srcpkgs/mstflint/template]: you can do ${version/%.3/-3}

File: srcpkgs/mstflint/template [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'mstflint'
+pkgname=mstflint
+version=4.11.0.3
+revision=1
+wrksrc="${pkgname}-4.11.0"
+build_style=gnu-configure
+configure_args="--enable-fw-mgr --enable-openssl"
+makedepends="boost-devel libcurl-devel libibmad-devel liblzma-devel \
```
Comment (maxice8) [inline srcpkgs/mstflint/template]: You are inside a quoted variable, you don't need to use backslash to denote a new line

File: srcpkgs/mstflint/template [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	libressl-devel libxml2-devel rdma-core-devel zlib-devel"
```
Comment (maxice8) [inline srcpkgs/mstflint/template]: please use 1 whitespace prefix on newlines being derived from a line being cut for too long

**Comment by Chocimier** [comment]
`wrksrc` and `distfiles` can be done update-proof like https://github.com/void-linux/void-packages/blob/master/srcpkgs/ebtables/template#L5 or https://github.com/void-linux/void-packages/pull/8792/files#diff-f4abc3aa30271142e8f3c8459f1c2622R17

**Comment by newbluemoon** [comment]
The cross builds fail because of hard-coded include-paths for libxml2 and curl in the `configure` script.
Patching `configure` like e.g.
```
pre_configure() {
	vsed -e "s;\(/usr/include/libxml2\);${XBPS_CROSS_BASE}\1;" \
	 -e "s;\(CURL_INC_DIR=\)\(/usr/include/curl\);\1${XBPS_CROSS_BASE}\2;" \
	 -i configure
}
```
works for me.

**Comment by newbluemoon** [comment]
The musl builds finish successfully by adding `CFLAGS+=" -D_GNU_SOURCE"` to the template and the following patch:
```
--- mtcr_ul/mtcr_ul_com.c.orig
+++ mtcr_ul/mtcr_ul_com.c
@@ -76,7 +76,7 @@

 #if CONFIG_ENABLE_MMAP
 #include <sys/mman.h>
-#include <sys/pci.h>
+#include <linux/pci.h>
 #include <sys/ioctl.h>
 #endif
```
All glibc’s `sys/pci.h` does is to include `linux/pci.h`.

**Comment by ArmedAviator** [comment]
Included musl fixes suggested by newbluemoon.

**Pull request closed without merging**